### PR TITLE
VTJ access tests

### PR DESCRIPTION
**Pull request by reynders**

### CODE REVIEW

File: service/src/integrationTest/kotlin/fi/espoo/evaka/pis/service/MockVtjClientService.kt [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import fi.espoo.evaka.shared.dev.DevPerson
+import fi.espoo.evaka.vtjclient.service.vtjclient.IVtjClientService
+import fi.espoo.evaka.vtjclient.soap.VTJHenkiloVastaussanoma
+import java.util.concurrent.ConcurrentHashMap
+
+class MockVtjClientService : IVtjClientService {
+
+    companion object {
+        private var queryRequestResponse:
+            ConcurrentHashMap<
```
Comment (Wnt) [inline service/src/integrationTest/kotlin/fi/espoo/evaka/pis/service/MockVtjClientService.kt:16]: Why not use normal HashMaps for `queryRequestResponse` and `queryCounts`? I don't think there's any situation where these will be accessed from multiple threads at the same time.

File: service/src/integrationTest/kotlin/fi/espoo/evaka/pis/service/MockVtjClientService.kt [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+        private var queryCounts:
+            ConcurrentHashMap<Pair<String, IVtjClientService.RequestType>, Int> =
+            ConcurrentHashMap()
+
+        fun resetQueryCounts() {
+            this.queryCounts.clear()
+        }
+
+        fun addPERUSSANOMA3Request(person: DevPerson) {
```
Comment (Wnt) [inline service/src/integrationTest/kotlin/fi/espoo/evaka/pis/service/MockVtjClientService.kt:29]: I would rename`addPERUSSANOMA3Request` and `addHUOLTAJAHUOLLETTAVARequest` into `add...Expectation`

File: service/src/integrationTest/kotlin/fi/espoo/evaka/pis/service/MockVtjClientService.kt [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                Pair<String, IVtjClientService.RequestType>, VTJHenkiloVastaussanoma.Henkilo
+            > =
+            ConcurrentHashMap()
+
+        private var queryCounts:
+            ConcurrentHashMap<Pair<String, IVtjClientService.RequestType>, Int> =
+            ConcurrentHashMap()
+
+        fun resetQueryCounts() {
+            this.queryCounts.clear()
```
Comment (Wnt) [inline service/src/integrationTest/kotlin/fi/espoo/evaka/pis/service/MockVtjClientService.kt:26]: I'd rename this to `resetMocks` and also clear `queryRequestResponse` in here

File: service/src/integrationTest/kotlin/fi/espoo/evaka/pis/service/MockVtjClientService.kt (hunk 1 above)
Comment (reynders) [reply service/src/integrationTest/kotlin/fi/espoo/evaka/pis/service/MockVtjClientService.kt:16]: This was requested in a PR review of the earlier implementation

**Review by akheron — APPROVED** [review]

**Pull request merged**